### ISSUE TITLE
Error when returning an array on render prop

### ISSUE THREAD

**Issue by davbrito** [diegomura/react-pdf#1872]
**Describe the bug**

When I try something like:
```jsx
<View
  render={({ pageNumber }) => (
    items.map(item => <Text>{item.label}, page: {pageNumber}</Text>)
  )}
/>
```

React Pdf throws an exception

**To Reproduce**
Steps to reproduce the behavior including code snippet (if applies):
1. Create a document with react PDF
2. Pass the render prop and return an array from there
3. It throwa

Codesanbox showing the errro: https://codesandbox.io/s/objective-greider-i5p544

**Expected behavior**
It should work normally as if the array was passed as children.

**Screenshots**
![image](https://user-images.githubusercontent.com/39559632/172206043-60b5fb1a-a4a8-4126-9522-24769df57d2b.png)

**Desktop (please complete the following information):**
 - OS: Windows and Linux
 - Browser Chrome
 - React-pdf version v2.1.2

**Comment by davbrito** [comment]
I also can't return a fragment or anything other than a react-pdf primitive.

**Comment by jasnross** [comment]
I was also running into this problem and came up with a potential fix in #2056
